### DISCUSSION TITLE
nz-tree prevent drop on nodes

### DISCUSSION THREAD

**Discussion by egalley** [NG-ZORRO/ng-zorro-antd#6959] [Q&A]
Hello,

Is there a way to prevent a node to be dropped on specific nodes (nzBeforeDrop?) and then to display a forbidenn sign and not to trigger nzOnDragEnd on that case?

Thank you for your help or suggestion

**Comment by yhornik** [comment]
Hey it's about Treeview. I don't know what is the type of data exactly. Sometimes it's NzTreeNode, But sometime NzTreeNodeOption

**Comment by egalley** [comment]
Hey,

It's solved.
I used nz-tree nzBeforeDrop, nzOnDragStart, nzOnDragOver and nzOnDragEnd.
Thank you
